TODO.md

```diff
@@ -8,27 +8,75 @@ see CHANGELOG.md for a more formal list of changes by release
 
 ## todo
 
-* gui, download progress bar *inside* the grid ...?
-    - pure fantasy?
-    - defer until after job queue
-
 * greater parallelism
     - internal job queue
     - replace log at bottom of screen with a list of jobs being processed and how far along they are
         - each job can be cancelled/stopped/discarded
+            - https://docs.oracle.com/javase/8/javafx/api/javafx/concurrent/Task.html
+            - https://clojuredocs.org/clojure.core/future-cancel
+        - a set of tasks can yield a compound 'job' with it's own progress that is the overall progress of the total
+        - the number of concurrent jobs can be controlled
+            - by default limit it to core count?
+
+---
+
+- ordered map of jobs
+- each job is clojure code
+- jobs can be cancelled
```

Comment:
technically yeah, with future-cancel, but I don't have a need for that right now.

I do need to look into `agent-shutdown` though.